schema/gen/go/genStruct.go

```diff
@@ -507,7 +507,9 @@ func (g structBuilderGenerator) emitMapAssemblerMethods(w io.Writer) {
 			case maState_finished:
 				panic("invalid state: Finish cannot be called on an assembler that's already finished")
 			}
-			//FIXME check if all required fields are set
+			if ma.s & fieldBits__{{ $type | TypeSymbol }}_sufficient != fieldBits__{{ $type | TypeSymbol }}_sufficient {
+				panic("invalid state: Not all required fields set")
```

Comment:
This should be an error return, not a panic.  Panics are reserved for when the builder has been used in a clearly invalid way (double finish, fed two keys in a row, things like that).  Data that's wanted but not present is just an error.

Bonus points if the error message can say which of the required fields are missing.